Formula/lilypond.rb

```diff
@@ -0,0 +1,66 @@
+class Lilypond < Formula
+  desc "Music engraving program"
+  homepage "https://lilypond.org"
+  url "https://lilypond.org/download/sources/v2.22/lilypond-2.22.1.tar.gz"
+  sha256 "72ac2d54c310c3141c0b782d4e0bef9002d5519cf46632759b1f03ef6969cc30"
+  license "GPL-3.0-or-later"
+
+  livecheck do
+    url "https://lilypond.org/source.html"
+    regex(/href=.*?lilypond[._-]v?(\d+(?:\.\d+)+)\.t/i)
+  end
+
+  depends_on "bison" => :build # Lilypond requires bison 2.4.1 or above
+  depends_on "fontforge" => :build
+  depends_on "gettext" => :build
+  depends_on "glib" => :build
+  depends_on "perl" => :build
+  depends_on "pkg-config" => :build
+  depends_on "t1utils" => :build
+  depends_on "texinfo" => :build # lilypond requires texinfo 6.1 or above
+  depends_on "texlive" => :build
+
+  depends_on "fontconfig"
+  depends_on "freetype"
+  depends_on "ghostscript"
+  depends_on "guile@2"
+  depends_on "pango"
+  depends_on "python@3.9"
+
+  uses_from_macos "flex" => :build
+
+  def install
+    # needed to solve conflicts with emacs
+    inreplace "config.make.in",
```

Comment:
Is there no `configure` flag we can use to customise this?

Reply:
I searched for lilypond source and I cannot find a flag to configure this directory.

Reply:
After some testing, I found a solution by changing the `datadir` option in configure. I'm making the changes and will send you the solution in a moment.

Reply:
I uploaded a change, but apparently it didn't work. There definitely doesn't seem to be a simple solution, as `make install` is creating links in `/usr/local/share`, even if you enter another `prefix`. I'll study this further and maybe get in touch with the developers for help.

Reply:
OK, I think I understand now. I'm not very familiar with `emacs`, but as I've seen in other formulas, like `libidn` and `clisp`, elisp files are linked to `/usr/local/share/emacs/site-lisp`. The correct placement that the homebrew rules suggest is to place these links in `/usr/local/share/emacs/site-lisp/{formula-name}` to avoid conflicts with other formulas. But lilypond install creates these links directly on the `site-lisp`. I'll find a solution to put it in the right place without making substitutions to `config.make.in`.

Reply:
tried: 
```
"--with-lispdir=#{elisp}"
```
without success: 
```
configure: WARNING: unrecognized options: --with-lispdir
```

Reply:
You can keep the `--datadir` option and apply the suggestion I made below.